ui/apps/dashboard/src/components/Runs/EnvRuns.tsx

```diff
@@ -0,0 +1,283 @@
+'use client';
+
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { RunsPage } from '@inngest/components/RunsPage/RunsPage';
+import type { Run } from '@inngest/components/RunsPage/types';
+import { useCalculatedStartTime } from '@inngest/components/hooks/useCalculatedStartTime';
+import {
+  useSearchParam,
+  useStringArraySearchParam,
+} from '@inngest/components/hooks/useSearchParam';
+
+import { useEnvironment } from '@/components/Environments/environment-context';
+import { useGetRun } from '@/components/RunDetails/useGetRun';
+import { useGetTraceResult } from '@/components/RunDetails/useGetTraceResult';
+import { useGetTrigger } from '@/components/RunDetails/useGetTrigger';
+import { graphql } from '@/gql';
+import { RunsOrderByField } from '@/gql/graphql';
+import { useCancelRun } from '@/queries/useCancelRun';
+import { useRerun } from '@/queries/useRerun';
+import { pathCreator } from '@/utils/urls';
+import { useGraphQLQuery, useSkippableGraphQLQuery } from '@/utils/useGraphQLQuery';
+import { usePlanFeatures } from '@/utils/usePlanFeatures';
+import { parseRunsData, toRunStatuses, toTimeField } from './utils';
+
+const GetRunsDocument = graphql(`
+  query GetEnvRuns(
+    $appIDs: [UUID!]
+    $environmentID: ID!
+    $startTime: Time!
+    $endTime: Time
+    $status: [FunctionRunStatus!]
+    $timeField: RunsOrderByField!
+    $functionRunCursor: String = null
+  ) {
+    environment: workspace(id: $environmentID) {
+      runs(
+        filter: {
+          appIDs: $appIDs
+          from: $startTime
+          until: $endTime
+          status: $status
+          timeField: $timeField
+        }
+        orderBy: [{ field: $timeField, direction: DESC }]
+        after: $functionRunCursor
+      ) {
+        edges {
+          node {
+            app {
+              externalID
+              name
+            }
+            function {
+              name
+              slug
+            }
+            id
+            queuedAt
+            endedAt
+            startedAt
+            status
+          }
+        }
+        pageInfo {
+          hasNextPage
+          hasPreviousPage
+          startCursor
+          endCursor
```

Comment:
**question (non-blocking)**: Would you recommend using fragments in the future?

Reply:
I think we should use a single query between both